test/interpreter_functional/test_suites/run_pipeline/esaggs_topmetrics.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import expect from '@kbn/expect';
+import { ExpectExpression, expectExpressionProvider } from './helpers';
+import { FtrProviderContext } from '../../../functional/ftr_provider_context';
+
+export default function ({
+  getService,
+  updateBaselines,
+}: FtrProviderContext & { updateBaselines: boolean }) {
+  let expectExpression: ExpectExpression;
+
+  describe('esaggs_topmetrics', () => {
+    before(() => {
+      expectExpression = expectExpressionProvider({ getService, updateBaselines });
+    });
+
+    const timeRange = {
+      from: '2015-09-21T00:00:00Z',
+      to: '2015-09-22T00:00:00Z',
+    };
+
+    describe('aggTopMetrics', () => {
+      it('can execute aggTopMetrics', async () => {
+        const expression = `
+          kibana_context timeRange={timerange from='${timeRange.from}' to='${timeRange.to}'}
+          | esaggs index={indexPatternLoad id='logstash-*'}
+          aggs={aggTerms id="1" enabled=true schema="bucket" field="extension.raw"}
+          aggs={aggTopMetrics id="2" enabled=true schema="metric" field="bytes" sortField="@timestamp" sortOrder="desc" size=3 }
+        `;
+        const result1 = await expectExpression('aggTopMetrics', expression).getResponse();
+
+        expect(result1.rows).to.eql([
+          { 'col-0-1': 'jpg', 'col-1-2': [1543, 7619, 8125] },
+          { 'col-0-1': 'css', 'col-1-2': [6573, 1770, 5077] },
+          { 'col-0-1': 'png', 'col-1-2': [1572, 6293, 4174] },
+          { 'col-0-1': 'gif', 'col-1-2': [602, 580, 101] },
+          { 'col-0-1': 'php', 'col-1-2': [7872, 9320, 257] },
+        ]);
+      });
+
+      it('can execute aggTopMetrics with different sortOrder and size', async () => {
+        const expression = `
+          kibana_context timeRange={timerange from='${timeRange.from}' to='${timeRange.to}'}
+          | esaggs index={indexPatternLoad id='logstash-*'}
+          aggs={aggTerms id="1" enabled=true schema="bucket" field="extension.raw"}
+          aggs={aggTopMetrics id="2" enabled=true schema="metric" field="bytes" sortField="@timestamp" sortOrder="asc" size=1 }
```

Comment:
One important property of top metrics is that terms is sortable by it. Could we have a test for that to make sure it works as expected?

Reply:
Added and fixed 👍